src/auth/auth.hpp

```diff
@@ -31,7 +32,36 @@ static const constexpr char *const kAllDatabases = "*";
  */
 class Auth final {
  public:
-  explicit Auth(const std::string &storage_directory);
+  struct Config {
+    Config() {}
+    Config(std::string name_regex, std::string password_regex, bool password_permit_null)
+        : name_regex_str{std::move(name_regex)},
+          password_regex_str{std::move(password_regex)},
+          password_permit_null{password_permit_null},
+          custom_name_regex{name_regex_str != glue::kDefaultUserRoleRegex},
+          custom_password_regex{password_regex_str != glue::kDefaultPasswordRegex} {}
+
+    std::string name_regex_str{glue::kDefaultUserRoleRegex};
+    std::string password_regex_str{glue::kDefaultPasswordRegex};
+    bool password_permit_null{true};
+
+   private:
+    friend class Auth;
+    bool custom_name_regex{false};
+    bool custom_password_regex{false};
```

Comment:
Can we prebuild the regex here, building regex is costly compared to using a regex.

Reply:
You're right, but gonna keep the string as well, since there are some logs and error messages that rely on it.